### ISSUE TITLE
Must use 8.3 format with cmd /c when concatenating args

### ISSUE THREAD

**Issue by robotdad** [Microsoft/vscode#9150]
- VSCode Version: 1.3.0
- OS Version: Win 10, 14367

Steps to Reproduce:
1. Setup a tasks.json with "cmd" and tasks that it calls to invoke an exe that requires additional params passed in args. Examples of one that works using 8.3 naming and one that doesn't using long paths below. 

``` json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "cmd",
    "args": ["/C"],
    "isShellCommand": true,
    "isBuildCommand": false,
    "suppressTaskName": true,
    "showOutput": "always",
    "tasks": [
        {
         "taskName": "Task (one lame)",
         "args":[
             "C:\\PROGRA~2\\MICROS~1.0\\Common7\\IDE\\devenv.exe",
             "/?"
         ],
         "taskName": "Task (two better but broken)",
         "args":[
             "C:/Program Files (x86)/Microsoft Visual Studio 14.0/Common7/IDE/devenv.exe",
             "/?"   
         ]
        }
    ]
}
```

**Comment by robotdad** [comment]
Here are some additional details based on what I'm actually trying to do. These are not easily run as they are using local paths on my machine, you need to setup Arduino IDE and have an Arduino Zero. So below is the output for each of the linked examples "Verify (Build)" task. 

While one of these examples does work in Code and the cmd prompt it doesn't unblock me as I have another task with a different command that also passes a lot of info as params, "Upload (Deploy to board)". It is failing as it isn't getting the paths in some of those args right even though they visibly appear so. Per the first example below, at least on Windows, something is different in how paths are interpreted when run by Code vs Windows. I've tried a lot of permutations of long and short paths with different escaping but not found one that works yet for this command. 

[tasks-buildtasklongwshortworks](https://gist.github.com/robotdad/b155bfd732f58eb9722b34ed09d8e7e9) (does not work in command prompt, works in VS Code)
running command> cmd /C C:/Program Files (x86)/Arduino/arduino-builder.exe -compile -logger=machine -hardware C:\PROGRA~2\Arduino\hardware -hardware C:\Users\mgoodner\AppData\Local\ARDUIN~1\packages -tools C:\PROGRA~2\Arduino\TOOLS-~1 -tools C:\PROGRA~2\Arduino\hardware\tools\avr -tools C:\Users\mgoodner\AppData\Local\ARDUIN~1\packages -built-in-libraries C:\PROGRA~2\Arduino\LIBRAR~1 -libraries D:\Source\Sketchbook\libraries -fqbn=arduino:samd:arduino_zero_edbg -ide-version=10608 -build-path d:\Source\Sketchbook\Blink\.vscode -warnings=none -prefs=build.warn_data_percentage=75 -verbose d:\Source\Sketchbook\Blink\Blink.ino

[tasks-buildtaskshortwshortworks](https://gist.github.com/robotdad/880d4d0f6f6f8fe2b77a69bc163ab4ee) (Works in both command prompt and VS Code)
running command> cmd /C C:\PROGRA~2\Arduino\ARDUIN~1.EXE -compile -logger=machine -hardware C:\PROGRA~2\Arduino\hardware -hardware C:\Users\mgoodner\AppData\Local\ARDUIN~1\packages -tools C:\PROGRA~2\Arduino\TOOLS-~1 -tools C:\PROGRA~2\Arduino\hardware\tools\avr -tools C:\Users\mgoodner\AppData\Local\ARDUIN~1\packages -built-in-libraries C:\PROGRA~2\Arduino\LIBRAR~1 -libraries D:\Source\Sketchbook\libraries -fqbn=arduino:samd:arduino_zero_edbg -ide-version=10608 -build-path d:\Source\Sketchbook\Blink\.vscode -warnings=none -prefs=build.warn_data_percentage=75 -verbose d:\Source\Sketchbook\Blink\Blink.ino

**Comment by dbaeumer** [comment]
Could you try the following. You actually don't need to specify cmd and the command if isShellCommand is true. You could do:

```json
{
   "command": "C:/Program Files (x86)/Microsoft Visual Studio 14.0/Common7/IDE/devenv.exe"
   "isShellCommand": true,
   ...
}
```
 which then knows better about escaping and how to pass that correctly to the OS.

**Comment by dbaeumer** [comment]
[Fixed in new terminal runner]
